src/planning/reflection_utils.ts

```diff
@@ -76,8 +76,8 @@ function getConstructorArgsAsTarget(
   // Take types to be injected from user-generated metadata
   // if not available use compiler-generated metadata
   let serviceIdentifier = serviceIdentifiers[index];
-  const injectIdentifier = (metadata.inject || metadata.multiInject);
-  serviceIdentifier = (injectIdentifier) ? (injectIdentifier) : serviceIdentifier;
+  const injectIdentifier = metadata.inject || metadata.multiInject;
+  serviceIdentifier = (injectIdentifier ? injectIdentifier : serviceIdentifier) as interfaces.ServiceIdentifier<unknown> | undefined;
```

Comment:
I know it was just simple type annotations being added here, but since `injectIdentifier` isn't used outside of the assignment on line 80, it would be great if someday we could rewrite this in the following manner:

```typescript
let serviceIdentifier = metadata.inject ?? metadata.multiInject ?? serviceIdentifiers[index];
```

Ideally this would avoid needing an explicit cast to `interfaces.ServiceIdentifier<unknown> | undefined`, since (I hope) we have `strictNullChecks` enabled.

This would be better enabled by making the return type of `formatTargetMetadata` use `interfaces.ServiceIdentifier<unknown>` for the `inject` and `multiInject` properties.